### ISSUE TITLE
Horrible Documentation

### ISSUE THREAD

**Issue by vin-fandemand** [google/googleapis.dart#230]
I was trying to use the Firestore google APIs for pure Dart server, and the documentation in the code is not at all helping. It's like it was written not to be used. I know this issue would be deleted but wanted to make sure someone would see it. Please let me know how can we help in adding better examples.

**Comment by kevmoo** [comment]
The documentation within each API is automatically generated from the source JSON. Not much we can do there.

Can you like to a specific case where something is confusing?

**Comment by vin-fandemand** [comment]
Recently I was working with Firestore API, and I wanted to write in batch. The document classes and methods did not help much I understand how it could be used. Then after digging into the package code, I realized that it is an implementation of [REST API](https://firebase.google.com/docs/firestore/reference/rest/v1/projects.databases.documents/batchWrite). This gave me better clarity. But I had to spend a whole day to understand what was missing and what was wrong.
